src/object.js

```diff
@@ -195,12 +217,13 @@ ReflectionObject.prototype.setParsedOption = function setParsedOption(name, valu
         });
         if (opt) {
             // If we found an existing option - just merge the property value
+            // (If it's a feature, will just write over)
```

Comment:
This feature really only gets called when parsing Root, since root gets initialized with defaults, and then can be overwritten if there's options specified at the root level. For example, if we start off with editions syntax, and then immediately add the following at the root level:

```
option features.json_format = LEGACY_BEST_EFFORT;
```

Since the previous experience is to merge at each child, then the features-specific behavior gets lost, since we'd end up with an array of values (`features.json_format = [LEGACY_BEST_EFFORT, ALLOW]`);

One option is to simply initialize defaults at the time that we set root-level options, so there's no overwriting, just 'filling in the blanks'. However this would require much of the same logic that `setParsedOption` already does (i.e., searching for a property within the`_proto_features` object), and it seems pretty un-DRY to rewrite it for features only because the setProperty value is different.

Instead, I'll change the `isFeature` parameter to `overwrite` to make it more general.

Reply:
This sounds related to my confusion about calling `setParsedOption` with defaults.  Alternatively, you could just initialize `_features` of the root to the relevant defaults during feature resolution right?  i.e. if `_parent` is null, initialize with the edition defaults

Reply:
Right, but what if `parent` isn't null? In that case we're rewriting a lot of the code that is in `setParsedOption` to confirm whether or not a value exists and if to rewrite it, or to create a new property, etc.